openlibrary/templates/books/add.html

```diff
@@ -63,6 +63,7 @@ <h1>$_("Add a book to Open Library")</h1>
                 <label for="id_name" class="hidden">$_("ID Type")</label>
                 <div id="id-errors" class="note hidden" data-i18n="$json_encode(i18n_strings)"></div>
                 <div id="confirm-add" class="note hidden">$_('ISBN may be invalid. Click "Add" again to submit.')</div>
+                <div id="id-found" class="note hidden"></div>
```

Comment:
```suggestion
```

Reply:
Thank you for the feedback! I've just about gotten everything reworked and ready to re-submit, but I ran into a slight wall. I reworked the `parseAndValidateIsbn10` and `parseAndValidateIsbn13` functions to add my async check after the isbn has been validated, but since it is async, Event.PreventDefault() isn't being triggered before it finishes submitting since the parseAndValidateIsbn functions aren't async. For context, this is the code I added and my updated fetch request.


<pre>
const isbnCheck =  async (event, idValue) => {
    const data = await checkIsbn(idValue);
    console.log(data);
    if (data !== null) {
        const error = 'An existing book was found with ISBN ${idValue}: ${data.title} - openlibrary.org/isbn/${idValue}';
        displayIsbnError(event, error);
    }
};
isbnCheck(event, idValue);
document.getElementById('id_value').value = idValue</pre>

<pre>
async function checkIsbn(idValue, callback=null) {
    let data = await fetch('/isbn/${idValue}.json?high_priority=false');
    let dataJson = await data.json();
    return dataJson;
}
</pre>

A second question I had, is it looks like for the other error messages, we're using error messages built within the template. For my error message, should I also be using that and then concatenating the ISBN/Book name?

Reply:
I'm so sorry for the massive delay! I thought that I had responded some time ago, but I somehow dropped the ball...

> For my error message, should I also be using that and then concatenating the ISBN/Book name?

You can a new error message to `i18n_strings`, [here](https://github.com/internetarchive/openlibrary/blob/e8ea2bacc63f491e619747437bc9b91006a222e8/openlibrary/templates/books/add.html#L5-L11).  Something like "A record for a book with this ISBN already exists." should work for now.


For the `parseAndValidateIsbnXX` methods/approach:

I think that something like this will work:

1. In `parseAndValidateId`, if the `fieldName` is `isbn_10` or `isbn_13`, make the `event.preventDefault()` call.  Remove this call from `displayIsbnError`.
2. Update `parseAndValidateIsbn10` and `parseAndValidateIsbn13` functions to return `false` if an error is detected, and `true` otherwise.  These functions don't need to be async.
3. In `parseAndValidateId`, if either `parseAndValidateIsbnXX` method returns `true`, get a reference to the form and call its [`submit()`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/submit) method to submit the form.